misc-utils/kill.c

```diff
@@ -603,14 +592,13 @@ static int kill_verbose(const struct kill_control *ctl)
 #ifdef USE_KILL_WITH_PIDFDINO
 		if ((ctl->pidfd_ino > 0)) {
 			int pfd;
-			pfd = pidfd_open(ctl->pid, 0);
-			if (pfd < 0)
-				err(EXIT_FAILURE, _("pidfd_open() failed: %d"), ctl->pid);
-
-			rc = validate_pfd_ino(pfd, ctl->pidfd_ino);
-			if (rc < 0)
+			pfd = ul_get_valid_pidfd(ctl->pid, ctl->pidfd_ino);
+			if (pfd < 0) {
+				if (errno == EBADF)
+					err(EXIT_FAILURE, _("pidfd_open() failed: %d"), ctl->pid);
 				errx(EXIT_FAILURE, _("pidfd inode %"PRIu64" not found for pid %d: %s"),
-								ctl->pidfd_ino, ctl->pid, strerror(-rc));
+								ctl->pidfd_ino, ctl->pid, strerror(errno));
```

Comment:
If there is a usable `errno`, then you can use `err()` rather than `errx()`, and it will append the error to the output, and you do not need strerror(errno).